docs/rules/no-array-callback-reference.md

```diff
@@ -7,6 +7,8 @@
 <!-- end auto-generated rule header -->
 <!-- Do not manually modify this header. Run: `npm run fix:eslint-docs` -->
 
+Passing functions to iterator methods can cause issues when the function is changed without realizing that the iterator passes 2 more parameters to it. **This also applies when using TypeScript,** albeit only if the function accepts the same parameter type used by the iterator method.
```

Comment:
This rule like `no-array-reduce` is very "controversial", so the clearer the better.